fencing/commands.c

```diff
@@ -1212,6 +1212,20 @@ stonith_device_remove(const char *id, gboolean from_cib)
         g_hash_table_remove(device_list, id);
         crm_info("Removed '%s' from the device list (%d active devices)",
                  id, g_hash_table_size(device_list));
+    } if (!device->cib_registered) {
```

Comment:
The "if" structure here needs to be tweaked to get all the cases right

 I think this test would be better as "if (from_cib)"